### PR TITLE
Add support for retrieving MIME type viewer configuration to oeq-ts-rest-api

### PR DESCRIPTION
**Pull request by edalex-ian**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included

##### Description of change

This PR Adds support the TS REST Module for the new endpoint added in #2339 to retrieve viewer configuration for a specified MIME type. This work is seen in `MimeType.ts` and `MimeType.test.ts`.

However when doing this I saw that the existing `/mimetype` query (to list MIME types) was still sitting in one of the front-end modules (`SearchFilterSettingsModule.ts`) so I refactored that out into the new `MimeType.ts` and then refactored the consumers as needed.

#1306

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: oeq-ts-rest-api/src/MimeType.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+   */
+  mimeType: string;
+  /**
+   * The description of a Mime type.
+   */
+  desc?: string;
+}
+
+/**
+ * Configuration for a MIME type viewer.
```
Comment (SammyIsConfused) [inline oeq-ts-rest-api/src/MimeType.ts:33]: Inconsistent capitalisation of MIME in the comments here.

File: oeq-ts-rest-api/src/MimeType.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+   */
+  config?: ViewerConfig;
+}
+
+/**
+ * The complete viewer configuration for a MIME type.
+ */
+export interface MimeTypeViewerConfiguration {
+  /**
+   * Which viewer should be used by default - .e.g. on search result pages.
```
Comment (SammyIsConfused) [inline oeq-ts-rest-api/src/MimeType.ts:102]:
```suggestion
   * Which viewer should be used by default - e.g. on search result pages.
```

**Review by SammyIsConfused — APPROVED** [review]
Looks good! Just a couple comment tweaks.

**Comment by edalex-ian** [comment]
Failed GHA builds are due to flaky tests - ApiDocsTest (new UI) and ItemSummary (Old UI). I'm going to do some of the feedback, so will wait for that to trigger a new build.

**Review by mrblippy — APPROVED** [review]
LGTM :t-rex:

**Comment by edalex-ian** [comment]
Cool, if I can get a :green_circle: build, I'll merge.

**Pull request merged**